syncctl.py

```diff
@@ -0,0 +1,272 @@
+        mirror_image(image_name_digest)
+    with open("work/images.tmp/mappings.json", "w") as f:
+        json.dump(digest_tag_mapping, f, indent=4, sort_keys=True)
+
+    if Path('work/images').is_dir():
+        shutil.rmtree("work/images")
+    os.rename("work/images.tmp", "work/images")
+
+def template_charts(api_versions: list[str], skip_charts: list[str], values: dict[str, str]) -> Generator[int, None, None]:
+    manifests = []
+    base_args = ["helm", "template"]
+    for api_version in api_versions:
+        base_args += ["--api-versions", api_version]
+    for k, v in values.items():
+        base_args += [f'--set={k}={v}']
+    for chart in list(Path('.').glob("work/helm-chart-repo/*.tgz")):
+        if chart.name in skip_charts:
+            debug(f"Skipping chart: {chart.name}")
+            continue
+        debug(f"Templating chart: {chart.name}")
+        p = subprocess.run(base_args + [chart], capture_output=True)
+        if p.returncode != 0:
+            if not "library charts are not installable" in str(p.stderr):
+                raise Exception(f'Error templating chart: {chart}, error: {p.stderr}')
+        yield p.stdout
+
+def extract_images(k8s_manifests: str) -> list[str]:
+    images = []
+    for d in yaml.load_all(k8s_manifests, Loader=yaml.SafeLoader):
+        # TODO: Handle kind: list + cephVersion
+        if d is None:
+            continue
+        if d["kind"] == "CronJob":
+            d["spec"] = d["spec"]["jobTemplate"]["spec"]
+        if d["kind"] in ["Deployment", "ReplicaSet", "StatefulSet", "DaemonSet", "Job", "CronJob", "ReplicationController"]:
+            containers = d["spec"]["template"]["spec"]["containers"]
+            if "initContainers" in d["spec"]["template"]["spec"]:
+                containers += d["spec"]["template"]["spec"]["initContainers"]
+            images += [container["image"] for container in containers]
+
+    # Special case Docker Hub
+    # https://github.com/containers/image/blob/1895e312af410ccdee5efa44e5223ec93ae76001/docs/containers-transports.5.md#dockerdocker-reference
+    for i, v in enumerate(images):
+        if "/" not in v:
+            images[i] = "docker.io/library/" + v
+        else:
+            s = v.split("/")
+            if not ("." in s[0] or ":" in s[0] or "localhost" == s[0]):
+                images[i] = "docker.io/" + v
+    return images
+
+def resolve_image(image: str) -> str:
+    p = subprocess.run(["skopeo", "inspect", "--raw", f'docker://{image}'], capture_output=True)
+    if p.returncode != 0:
+        raise Exception(f'Error inspecting image: {image}, error: {p.stderr}')
+
+    manifest = json.loads(p.stdout)
+    # https://github.com/opencontainers/image-spec/blob/43a7dee1ec31e0ad091d2dc93f6ada1392fba587/image-index.md
+    if manifest["mediaType"] == "application/vnd.docker.distribution.manifest.list.v2+json" or manifest["mediaType"] == "application/vnd.oci.image.index.v1+json":
```

Comment:
Split this over two lines
```suggestion
    if (manifest["mediaType"] == "application/vnd.docker.distribution.manifest.list.v2+json"
            or manifest["mediaType"] == "application/vnd.oci.image.index.v1+json"):
```